src/providers/flagsView.ts

```diff
@@ -113,16 +120,42 @@ export class LaunchDarklyTreeViewProvider implements vscode.TreeDataProvider<Fla
 
 	private async flagUpdateListener() {
 		// Setup listener for flag changes
-		this.flagStore.on('update', async flag => {
+		this.flagStore.on('update', async (keys: string) => {
 			try {
-				const updatedFlag = await this.api.getFeatureFlag(this.config.project, flag.key, this.config.env);
-				const updatedIdx = this.flagNodes.findIndex(v => v.flagKey === updatedFlag.key);
-				this.flagNodes[updatedIdx] = this.flagToValues(updatedFlag);
+				const flagKeys = Object.values(keys);
+				flagKeys.map(key => {
+					this.flagStore.getFeatureFlag(key).then(updatedFlag => {
+						const updatedIdx = this.flagNodes.findIndex(v => v.flagKey === key);
+						this.flagToValues(updatedFlag.flag, updatedFlag.config).then(newFlagValue => {
+							this.flagNodes[updatedIdx] = newFlagValue;
+						});
+					});
+				});
 				this.refresh();
 			} catch (err) {
 				console.error('Failed to update LaunchDarkly flag tree view:', err);
 			}
 		});
+		this.flagStore.storeUpdates.event(async () => {
+			const flags = await this.flagStore.allFlagsMetadata();
+			if (flags.length != this.flagNodes.length) {
```

Comment:
I'll have to look into eslint on why this isn't being caught, but we should always use strict equality (or not equal) checks. Can you pass through this changeset and make sure we'er always using `!==`?